### PR TITLE
Refactor paxcounter

### PR DESCRIPTION
**Pull request by thoherr**
Implements #3342 (using timesliced count, not yet configurable)

### CODE REVIEW

File: src/modules/esp32/PaxcounterModule.cpp [hunk 1]
```diff
@@ -31,9 +49,12 @@ bool PaxcounterModule::sendInfo(NodeNum dest)
     meshtastic_MeshPacket *p = allocDataProtobuf(pl);
     p->to = dest;
     p->decoded.want_response = false;
-    p->priority = meshtastic_MeshPacket_Priority_MIN;
+    p->priority = meshtastic_MeshPacket_Priority_DEFAULT;
```
Comment (caveman99) [inline src/modules/esp32/PaxcounterModule.cpp:52]: Please leave this as min, regular message traffic should have priority.

File: src/modules/esp32/PaxcounterModule.h [hunk 2]
```diff
@@ -8,11 +8,15 @@
 #include <libpax_api.h>
 
 /**
- * A simple example module that just replies with "Message received" to any message it receives.
+ * Wrapper module for the estimate passenger (PAX) count library (https://github.com/dbinfrago/libpax) which
+ * implements the core functionality of the ESP32 Paxcounter project (https://github.com/cyberman54/ESP32-Paxcounter)
  */
```
Comment (caveman99) [inline src/modules/esp32/PaxcounterModule.h:13]: Woops, good catch :-)

File: src/modules/esp32/PaxcounterModule.cpp [hunk 3]
```diff
@@ -76,12 +97,14 @@ int32_t PaxcounterModule::runOnce()
...
-            libpax_counter_init(NullFunc, &count_from_libpax, UINT16_MAX, 1);
+            libpax_counter_init(handlePaxCounterReportRequest, &count_from_libpax,
+                                moduleConfig.paxcounter.paxcounter_update_interval, 0);
             libpax_counter_start();
         } else {
             sendInfo(NODENUM_BROADCAST);
         }
-        return getConfiguredOrDefaultMs(moduleConfig.paxcounter.paxcounter_update_interval, default_broadcast_interval_secs);
+        // we check every second if the counter had new data to send
+        return 1000;
```
Comment (caveman99) [inline src/modules/esp32/PaxcounterModule.cpp:107]: this is rather heavy on the threading system, like polling for a flag. My suggestion is to keep the original interval (no change here) and trigger the osthread from the outside when new data arrived. See comment above

File: src/modules/esp32/PaxcounterModule.cpp [hunk 4]
```diff
@@ -7,21 +7,39 @@
...
+ * We only clear our sent flag here, since this function is called from another thread, so we
+ * cannot send to the mesh directly.
+ */
+void PaxcounterModule::handlePaxCounterReportRequest()
+{
+    // The libpax library already updated our data structure, just before invoking this callback.
+    LOG_INFO("PaxcounterModule: libpax reported new data: wifi=%d; ble=%d; uptime=%lu\n",
+             paxcounterModule->count_from_libpax.wifi_count, paxcounterModule->count_from_libpax.ble_count, millis() / 1000);
+    paxcounterModule->reportedDataSent = false;
+}
```
Comment (caveman99) [inline src/modules/esp32/PaxcounterModule.cpp:21]: you can trigger the osthread processing here by calling paxcounterModule->setIntervalFromNow(0);

**Review by caveman99 — CHANGES_REQUESTED** [review]

File: src/modules/esp32/PaxcounterModule.cpp (hunk 1 above)
Comment (thoherr) [reply src/modules/esp32/PaxcounterModule.cpp:52]: ok, no problem.
Just a question: When would BACKGROUND be appropriate?

Comment (caveman99) [reply src/modules/esp32/PaxcounterModule.cpp]: Pretty well explained in https://github.com/meshtastic/protobufs/blob/00332412b238fe559175a6e83fdf8d31fa5e209a/meshtastic/mesh.proto#L819-L878 ... I think BACKGROUND may be more appropriate than MIN here, and in other places that use it. (Telemetry and so on...) For now i'd stick to min cause the other plugins use min too. Let's address this in a different PR.

**Review by caveman99 — APPROVED** [review]

**Pull request merged**